facturacion_mexico/facturacion_fiscal/doctype/factura_fiscal_mexico/factura_fiscal_mexico.py

```
@@ -322,58 +322,133 @@ def _is_nota_descuento(self) -> bool:
+			)
+		)
+		if es_descuento:
+			return NOTA_CREDITO_DESCUENTO
+
+		# Ni devolución ni descuento exactos → NO clasificar por default; bloquear.
+		self._throw_estado_nc_invalido(
+			_(
+				"La Sales Invoice Return no coincide exactamente con un estado válido de devolución de "
+				"mercancía ni de descuento/bonificación; revísela antes de timbrar."
+			)
+		)
```

Ruff (0.16.0)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)